debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/transforms/UnwrapFromMongoDbEnvelopeTest.java

```diff
@@ -269,4 +275,217 @@ public void shouldGenerateRecordForDeleteEvent() throws InterruptedException {
 
         assertThat(value).isNull();
     }
+
+    @Test
+    public void shouldNotFlattenTransformRecordForInsertEvent() throws InterruptedException {
+        CollectionId collectionId = new CollectionId("rs0", "dbA", "c1");
+        BsonTimestamp ts = new BsonTimestamp(1000, 1);
+        ObjectId objId = new ObjectId();
+        Document address = new Document()
+                .append("street", "Morris Park Ave")
+                .append("zipcode", "10462");
+        Document obj = new Document().append("_id", objId)
+                .append("name", "Sally")
+                .append("address", address);
+
+        // given
+        Document event = new Document().append("o", obj)
+                                       .append("ns", "dbA.c1")
+                                       .append("ts", ts)
+                                       .append("h", Long.valueOf(12345678))
+                                       .append("op", "i");
+        RecordsForCollection records = recordMakers.forCollection(collectionId);
+        records.recordEvent(event, 1002);
+        assertThat(produced.size()).isEqualTo(1);
+        SourceRecord record = produced.get(0);
+
+        // when
+        SourceRecord transformed = transformation.apply(record);
+
+        Struct key = (Struct) transformed.key();
+        Struct value = (Struct) transformed.value();
+
+        // then assert key and its schema
+        assertThat(key.schema()).isSameAs(transformed.keySchema());
+        assertThat(key.schema().field("id").schema()).isEqualTo(SchemaBuilder.OPTIONAL_STRING_SCHEMA);
+        assertThat(key.get("id")).isEqualTo(objId.toString());
+
+        // and then assert value and its schema
+        assertThat(value.schema()).isSameAs(transformed.valueSchema());
+        assertThat(value.get("name")).isEqualTo("Sally");
+        assertThat(value.get("id")).isEqualTo(objId.toString());
+        assertThat(value.get("address")).isEqualTo(new Struct(value.schema().field("address").schema())
+            .put("street", "Morris Park Ave").put("zipcode", "10462"));
+
+        assertThat(value.schema().field("id").schema()).isEqualTo(SchemaBuilder.OPTIONAL_STRING_SCHEMA);
+        assertThat(value.schema().field("name").schema()).isEqualTo(SchemaBuilder.OPTIONAL_STRING_SCHEMA);
+        assertThat(value.schema().field("address").schema()).isEqualTo(
+                SchemaBuilder.struct()
+                    .name("serverX...dbA.c1.address")
```

Comment:
@jpechane That expected schema name looks weird, I'm going to explore whether it's a test only issue or something serious.

Reply:
Ok, so that was just caused by a slightly confusing server name in the test. I've adjusted it.